### PR TITLE
distutils and difflib stubtest exceptions

### PR DESCRIPTION
**Pull request by hatal175**
- set_python_build was deleted back in 2002 but never removed from docs
- FileList is not really documented so I just derived it from the code
- values for some arguments in filelist are int while actually used only for boolean checks. I left the int but added a bool with a Union because this is weird.
- Exception for SequenceMatcher.__init__ can be removed if https://github.com/python/mypy/issues/3737 is handled at some point.

### CODE REVIEW

File: stdlib/distutils/filelist.pyi [hunk 1]
```diff
@@ -1 +1,80 @@
-class FileList: ...
+from typing import Callable, Iterable, List, Optional, Pattern, Union, overload
+from typing_extensions import Literal
+
+# class is entirely undocumented
+class FileList:
+    allfiles: Optional[Iterable[str]] = ...
+    files: List[str] = ...
+    def __init__(
+        self, warn: Optional[Callable[[str], None]] = ..., debug_print: Optional[Callable[[str], None]] = ...
```
Comment (hauntsaninja) [inline stdlib/distutils/filelist.pyi:9]: It looks like these arguments are ignored, so I'd just keep their types as None

File: stdlib/distutils/filelist.pyi [hunk 2]
```diff
@@ -1 +1,80 @@
...
+        anchor: Union[int, bool] = ...,
+        prefix: Optional[str] = ...,
+    ) -> bool: ...
+    @overload
+    def include_pattern(
+        self,
+        pattern: Union[str, Pattern[str]],
+        anchor: Union[int, bool] = ...,
+        prefix: Optional[str] = ...,
+        is_regex: Literal[0, False] = ...,
```
Comment (hauntsaninja) [inline stdlib/distutils/filelist.pyi:32]: I think this is backwards; we allow pattern to be a compiled regex if `is_regex` is True. Also if is_regex is True, the other arguments are ignored, so we shouldn't allow them to be passed.

**Review by hauntsaninja — APPROVED** [review]
Thanks, this lgtm! I added some comments, but since the whole class is undocumented and distutils is deprecated, I don't feel strongly.

File: stdlib/distutils/filelist.pyi (hunk 2 above)
Comment (hatal175) [reply stdlib/distutils/filelist.pyi:30]: Fixed that one. I also added the is_regex True branch with positional args. Is that what you had in mind?

Comment (hauntsaninja) [reply stdlib/distutils/filelist.pyi:30]: Yup, that's what I had in mind :-) One more nit: I think the first overload of include_pattern / exclude_pattern is now redundant

**Comment by hauntsaninja** [comment]
Thanks!

**Pull request merged**